### ISSUE TITLE
Issue with file:// being disabled for unknown reasons by default.

### ISSUE THREAD

**Issue by FjicdeiwFeiej** [electron/electron#10815]
In Electron, "fetch('./assets/somedata.json')" caused the error:
'Fetch API cannot load file:///C:/ElectronTest/resources/app.asar/assets/somedata.json. URL scheme "file" is not supported.'

By adding the following code, which I found out by a happy accident for once, it started working. How come?

const { webFrame } = require('electron');
webFrame.registerURLSchemeAsPrivileged('file');

I mean, why would file:// be disabled in Electron? It makes no sense. The whole point of Electron is that it has privileges. I guess you "prefer" that we use the Node fs stuff? But still, why not also allow this by default? Not having to modify code in all sorts of existing code that uses fetch() seems sane to me.

**Comment by MarshallOfSound** [comment]
This is the same security policy as Chromium, it's that people can't request arbitrary things from your filesystem.  Calling `webFrame.registerURLSchemeAsPriviledged('file')` is a bad plan in terms of security, if you load any remote JS or remote content it's going to be a security nightmare.

I'd strongly recommend using node's `fs` module instead of using `fetch` and the `file:` protocol.

TLDR: Security

**Comment by FjicdeiwFeiej** [comment]
But the existing code uses fetch, so it has to be actively changed. Even though it was my code this time. Just seems bizarre to allow fs but not fetch.

**Comment by MarshallOfSound** [comment]
@FjicdeiwFeiej Enabling / disabinling nodeIntegration is a user choice if they want access to that power in the renderer process.  If we blanket allow `file://` fetches it violates existing security models.  You code that uses fetch will traditionally be hosted on an HTTP server and therefore work, in Electron it's local `file://` so you'll have to use different tech

**Comment by cakekindel** [comment]
Sorry to hijack an old thread but what would a safe alternative for loading a user's local files in my app be?
